<p><center><br />
<img src="/usr/src/hic_qc/collateral/PGBlueLogoHorSmall.png" alt="Phase Genomics logo" width="200" class="center"></p>
<h1>Hi-C Library QC Report</h1>
<h2>Input Data</h2>
<table>
<thead>
<tr>
<th style="text-align: left;">File Type &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</th>
<th style="text-align: left;">File Name &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">BAM File</td>
<td style="text-align: left;">P_citronellum.asm.hic.hap1.p_ctg_vs_HiC.sorted.bam</td>
</tr>
<tr>
<td style="text-align: left;">Assembly File</td>
<td style="text-align: left;">P_citronellum_OmniC.hic.bp.hap1.p_ctg.fa</td>
</tr>
<tr>
<td style="text-align: left;">Forward Hi-C Reads</td>
<td style="text-align: left;"><span class="mixed-results">forward Hi-C reads not found</span></td>
</tr>
<tr>
<td style="text-align: left;">Reverse Hi-C Reads</td>
<td style="text-align: left;"><span class="mixed-results">reverse Hi-C reads not found</span></td>
</tr>
</tbody>
</table>
<h2>Genome Scaffolding Sufficiency</h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Label</th>
<th style="text-align: center;">Library statistics</th>
<th style="text-align: right;">Expected values</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">Subjective Hi-C library judgment</td>
<td style="text-align: center;"><span class="pass">SUFFICIENT</span></td>
<td style="text-align: right;">See Judgment</td>
</tr>
<tr>
<td style="text-align: left;">Same strand high-quality* (HQ) read pairs (RPs)</td>
<td style="text-align: center;"><span class="pass">14.48%</span></td>
<td style="text-align: right;">&gt; 1.5%</td>
</tr>
<tr>
<td style="text-align: left;">Informative RPs**</td>
<td style="text-align: center;"><span class="pass">13.40%</span></td>
<td style="text-align: right;">&gt; 5.0%</td>
</tr>
</tbody>
</table>
<div class="small center">
<br />
*High-quality (HQ) read pairs have minimum mapping quality >= 20, maximum edit distance <= 5, and are not duplicates.<br>
**Informative read pairs are read pairs which have MAPQ > 0, are not PCR duplicates, and map to different contigs or >10kb apart.

</div>

<h2>Metrics Demonstrating Strong Proximity Signal</h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Label</th>
<th style="text-align: center;">Library statistics</th>
<th style="text-align: right;">Expected values</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">Fraction of HQ RPs &gt;10kb apart (CTGs &gt;10kb)*</td>
<td style="text-align: center;"><span class="pass">13.59%</span></td>
<td style="text-align: right;">&gt; 3.0%</td>
</tr>
<tr>
<td style="text-align: left;">Fraction of HQ RPs Intercontig (CTGs &gt;10kb)**</td>
<td style="text-align: center;"><span class="pass">28.42%</span></td>
<td style="text-align: right;">&gt; 2.5%</td>
</tr>
<tr>
<td style="text-align: left;">Informative HQ RPs per-contig per-1M-RPs (CTGs &gt;5kb)***</td>
<td style="text-align: center;"><span class="fail">40.25</span></td>
<td style="text-align: right;">&gt; 100.0</td>
</tr>
</tbody>
</table>
<div class="small center">
<br />
*The proportion of <em>read pairs that span at least 10kb</em>, out of all read pairs that map (a) with high-quality, (b) to the same contig, (c) where that contig is at least 10kb long.<br>
**The proportion of <em>read pairs mapping to two different contigs each greater than 10kb</em>, out of all read pairs that map with high-quality.<br>
***The average number of HQ reads, per contig <em>among contigs at least 5kb in length</em>, per million read pairs, that are informative for identifying contigs which belong on the same chromosome.<br>
</div>

<h2>Noninformative Read Pair Breakdown</h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Label</th>
<th style="text-align: center;">Library statistics</th>
<th style="text-align: right;">Expected values</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">Noninformative RPs*</td>
<td style="text-align: center;"><span class="fail">67.80%</span></td>
<td style="text-align: right;">&lt;= 50.0%</td>
</tr>
<tr>
<td style="text-align: left;">Duplicate reads</td>
<td style="text-align: center;"><span class="pass">0.00%</span></td>
<td style="text-align: right;">&lt; 20.0%</td>
</tr>
<tr>
<td style="text-align: left;">Zero map distance read pairs</td>
<td style="text-align: center;"><span class="pass">0.34%</span></td>
<td style="text-align: right;">&lt;= 20.0%</td>
</tr>
<tr>
<td style="text-align: left;">Zero MAPQ reads</td>
<td style="text-align: center;"><span class="fail">60.78%</span></td>
<td style="text-align: right;">&lt;= 20.0%</td>
</tr>
<tr>
<td style="text-align: left;">Unmapped reads</td>
<td style="text-align: center;"><span class="pass">0.00%</span></td>
<td style="text-align: right;">&lt;= 10.0%</td>
</tr>
</tbody>
</table>
<div class="small center">
<br />
*Note that the sum of informative and noninformative read pairs is not 100% because read pairs with mapping distance between 1b and 10kb are not classified as either informative or noninformative.<br>
Because noninformative reads can belong to more than one category, these numbers may sum to a value larger than the overall noninformative read pair amount at the top of the report.<br>
</div>

<div class="pagebreak"> </div>

<h2>Assembly Statistics</h2>
<table>
<thead>
<tr>
<th style="text-align: left;">Label</th>
<th style="text-align: right;">Assembly statistics</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">BAM file</td>
<td style="text-align: right;">P_citronellum.asm.hic.hap1.p_ctg_vs_HiC.sorted.bam</td>
</tr>
<tr>
<td style="text-align: left;">Forward Hi-C Reads</td>
<td style="text-align: right;"><span class="mixed-results">forward Hi-C reads not found</span></td>
</tr>
<tr>
<td style="text-align: left;">Reverse Hi-C Reads</td>
<td style="text-align: right;"><span class="mixed-results">reverse Hi-C reads not found</span></td>
</tr>
<tr>
<td style="text-align: left;">Assembly file</td>
<td style="text-align: right;">P_citronellum_OmniC.hic.bp.hap1.p_ctg.fa</td>
</tr>
<tr>
<td style="text-align: left;">Assembly size</td>
<td style="text-align: right;">514,186,233</td>
</tr>
<tr>
<td style="text-align: left;">Contig (CTG) N50</td>
<td style="text-align: right;">760,978</td>
</tr>
<tr>
<td style="text-align: left;">CTGs</td>
<td style="text-align: right;">1,566</td>
</tr>
<tr>
<td style="text-align: left;">CTGs &gt; 10kb</td>
<td style="text-align: right;">1,566</td>
</tr>
<tr>
<td style="text-align: left;">CTGs &gt; 5kb</td>
<td style="text-align: right;">1,566</td>
</tr>
</tbody>
</table>
<h2>Extended Library Statistics</h2>
<p><center></p>
<table>
<thead>
<tr>
<th style="text-align: left;">Label</th>
<th style="text-align: right;">Library statistics</th>
<th style="text-align: right;">Expected values</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align: left;">Total read pairs (RPs) analyzed</td>
<td style="text-align: right;">2,000,001</td>
<td style="text-align: right;">N/A</td>
</tr>
<tr>
<td style="text-align: left;">High-quality (HQ) RPs</td>
<td style="text-align: right;">22.18%</td>
<td style="text-align: right;">N/A</td>
</tr>
<tr>
<td style="text-align: left;">Clustering usable HQ reads per contig (CTGs &gt;5kb)*</td>
<td style="text-align: right;"><span>80.50</span></td>
<td style="text-align: right;">&gt; 600.0</td>
</tr>
<tr>
<td style="text-align: left;">RPs &gt;10kb apart</td>
<td style="text-align: right;">9.79%</td>
<td style="text-align: right;">1-15%</td>
</tr>
<tr>
<td style="text-align: left;">RPs &gt;10kb apart (CTGs &gt;10kb)</td>
<td style="text-align: right;">25.77%</td>
<td style="text-align: right;">1-15%</td>
</tr>
<tr>
<td style="text-align: left;">Intercontig RPs</td>
<td style="text-align: right;">62.03%</td>
<td style="text-align: right;">10-60% (contigs) 1-20% (chromosomes)</td>
</tr>
<tr>
<td style="text-align: left;">Intercontig HQ RPs</td>
<td style="text-align: right;">28.42%</td>
<td style="text-align: right;">10-60% (contigs) 1-20% (chromosomes)</td>
</tr>
<tr>
<td style="text-align: left;">Same strand RPs</td>
<td style="text-align: right;">22.92%</td>
<td style="text-align: right;">2-50%</td>
</tr>
<tr>
<td style="text-align: left;">Split reads</td>
<td style="text-align: right;">19.56%</td>
<td style="text-align: right;">1-10% (PG libraries) 30%+ (other libraries)</td>
</tr>
<tr>
<td style="text-align: left;">Alignment Parameters</td>
<td style="text-align: right;">bwa mem -t 24 -5SP</td>
<td style="text-align: right;">N/A</td>
</tr>
<tr>
<td style="text-align: left;">Samblaster Parameters</td>
<td style="text-align: right;">samblaster command not found</td>
<td style="text-align: right;">N/A</td>
</tr>
<tr>
<td style="text-align: left;">Restriction Enzyme(s)</td>
<td style="text-align: right;">unspecified</td>
<td style="text-align: right;">N/A</td>
</tr>
<tr>
<td style="text-align: left;"></center></td>
<td style="text-align: right;"></td>
<td style="text-align: right;"></td>
</tr>
</tbody>
</table>
<div class="small center">
<br />
*The average number of usable high-quality read pairs per contig, for contigs greater than 5kb. Read pairs are "usable" if they map (a) with high-quality, (b) to different contigs, (c) where each of those contigs are greater than 5kb and (d) both mappings are high-quality.<br>
</div>

<div class="pagebreak"> </div>

<h2>Aligned mate distance histograms</h2>
<p><img alt="&quot;Long range interaction histogram&quot;" src="/scratch/srb67793/Pelargonium/Results/HiFiASM_OmniC/P_citronellum.hap1_vs_HiC_long.png" /><br />
<img alt="&quot;Short range interaction histogram&quot;" src="/scratch/srb67793/Pelargonium/Results/HiFiASM_OmniC/P_citronellum.hap1_vs_HiC_short.png" /></p>
<p><img alt="&quot;Log-log interaction histogram (counts)&quot;" src="/scratch/srb67793/Pelargonium/Results/HiFiASM_OmniC/P_citronellum.hap1_vs_HiC_log_log.png" /><br />
<img alt="&quot;Log-log interaction histogram (density)&quot;" src="/scratch/srb67793/Pelargonium/Results/HiFiASM_OmniC/P_citronellum.hap1_vs_HiC_log_log_norm.png" /></p>
<div class="pagebreak"> </div>
<h2>Alignment distance statistics and plots</h2>
<p>We briefly describe some of the statistics we compute below to aid interpretation of this report.</p>
<h3>Subjective Hi-C Library Judgment</h3>
<p>While Hi-C data is nuanced and some analyses are more sensitive to data quality than others, a basic quality assessment can usually be made by examining the mapping characteristics of the Hi-C library. Based on our experience working with Hi-C data, we classify libraries into one of four QC categories:<br />
 - <strong>Sufficient</strong> means that from everything we can tell, the library looks to be in great shape. Proceed to full sequencing or analysis with confidence.<br />
 - <strong>Mixed Results</strong> means that the library is good in some ways, but not in others. Perhaps it has a good amount of long-range data, but there are also an elevated number of read pairs with MAPQ 0. Usually, data generated from Mixed Results libraries works out just fine (a high MAPQ 0 number can be due to repetitiveness in the assembly or unpurged haplotigs, for example), but it is good to know there may have been a few hiccups in the library prep in case troubleshooting is needed down the line.<br />
 - <strong>Low Signal</strong> means that the library contains good Hi-C signal, but it's in lower percentage than usual. These libraries are generally good for generating useful Hi-C data, but you may need to sequence a little deeper than normal to get enough of it. You might consider size selecting the library to discard reads outside the 300-700bp range, as these are unlikely to be good Hi-C junctions. Alternatively, you might just want to prep a new library.<br />
 - <strong>Insufficient</strong> means that the library, or perhaps the library in combination with a low-contiguity or error-prone assembly, does not look useful. Sometimes size selection can rescue such libraries, but sometimes a new prep is the only way forward. <br><br><a href="mailto:support@phasegenomics.com">Contact us</a> if your library is not sufficient and we will help you out.</p>
<h3>Read pairs on same strand</h3>
<p>This is the percentage of reads mapping to the same contig in the same orientation (FF or RR). For shotgun libraries, this should be ~1%, but for a pure Hi-C library, it could be as high as 50%. This is a primary metric of Hi-C library quality because it is minimally affected by assembly contiguity.</p>
<h3>Read pairs &gt; 10kb apart</h3>
<p>This is the percentage of read pairs which map to the same contig, with at least 10kb separating them. More is always better, but because this number is affected by assembly contiguity, there is not a specific target threshold. Note that for some analyses, such as scaffolding or metagenomic deconvolution, read pairs that map to the same contig are not useful because they do not provide information that the assembly doesn't already contain. This statistic is more useful for these projects because it correlates with library prep success. These reads are useful for analyses like structural variant analysis or assembly misjoin detection, because they provide detailed structural information about existing assembled sequences.</p>
<h3>Read pairs &gt; 10kb apart mapping to contigs &gt;10kb</h3>
<p>This is the percentage of read pairs which map to the same contig, with at least 10kb separating them, but only considering read pairs mapping to contigs that are at least 10kb long. This attempts to corrects for assembly contiguity differences. More is always better, but typically at least 5% is desired. Note that for some analyses, such as scaffolding or metagenomic deconvolution, read pairs that map to the same contig are not useful because they do not provide information that the assembly doesn't already contain. This statistic is more useful for these projects because it correlates with library prep success. These reads are useful for analyses like structural variant analysis or assembly misjoin detection, because they provide detailed structural information about existing assembled sequences.</p>
<h3>Read pairs mapping to different contigs or chromosomes</h3>
<p>This is the percentage of read pairs which map to different contigs, which is particularly important. More is always better, but because this number is affected by assembly quality, there is not a specific target threshold, although at least 20% on <em>de novo</em> assembly projects is helpful. These reads are the primary source of information for Hi-C scaffolding or metagenomic deconvolution analyses. This statistic useful on most <em>de novo</em> projects because it also correlates with library prep success. These reads may be useful for analyses like structural variant analysis or assembly misjoin detection if those are performed on lower contiguity assemblies, because they provide detailed structural information about sequences which were not assembled together into contigs.</p>
<h3>Split reads</h3>
<p>Traditionally, split reads have been a favored measure of Hi-C library quality because they directly exhibit Hi-C junctions. Most traditional Hi-C library preparations produce many reads that sequence through junctions because their Hi-C junctions tend to occur randomly on the proximity ligated chimeric molecules. However, Phase Genomics libraries, whether produced in our laboratory or by means of our Plant, Animal, Human, or Microbe Hi-C kits, will have a generally lower percentage of split reads. This is because we have optimized our Hi-C protocol to enrich for slightly longer fragments around Hi-C junctions, such that each read is less likely to read through a junction even when a junction is present.</p>
<p>This innovation improves mappability and increases the amount of useful data and reduces the utility of split read measurements to assess library quality. We therefore rely more heavily on metrics that directly relate to the usefulness of Hi-C reads for proximity analysis, such as the percentage of read pairs that map to the same strand or to different contigs.</p>
<h3>Duplicate reads</h3>
<p><strong>IMPORTANT NOTE: THE DUPLICATE FLAG IS NOT SET BY DEFAULT IN A BAM FILE. YOU NEED TO EXPLICITLY SET IT BY E.G. RUNNING SAMBLASTER OR PICARD MARKDUPLICATES ON YOUR BAM FILE. IF THE PERCENT OF DUPLICATES IS EXACTLY ZERO, IT PROBABLY MEANS THAT THE FLAG HAS NOT BEEN SET.</strong></p>
<p>Sequencing libraries frequently contain duplicate reads due to PCR or optical issues. These are generally considered to be non-informative because they are chemical artifacts rather than biological signal and are thus typically excluded from further analysis. Higher percentages of duplicate reads are also correlated with low library complexity and poor library performance, making the percentage of duplicate reads a useful quality control measure.</p>
<h3>Unmapped reads</h3>
<p>A high percent of unmapped reads may indicate sequence is missing from the reference, the reads are mapped to the wrong reference, or the sample is contaminated.</p>
<p><br></p>
<h4 class="centered">
REPORT VERSION: 0+untagged.261.g6881c33
</h4>